bgpd/bgp_evpn.c

```diff
@@ -1604,22 +1596,11 @@ static int update_evpn_route_entry(struct bgp *bgp, struct bgpevpn *vpn,
 	if (!local_ri && !add)
 		return 0;
 
-	if (!local_ri) {
-		/* When learnt locally for the first time but already known from
-		 * remote, we have to initiate appropriate MAC mobility steps.
-		 * This is applicable when updating the VNI routing table.
-		 * We need to skip mobility steps for g/w macs (local mac on g/w
-		 * SVI) advertised in EVPN.
-		 * This will ensure that local routes are preferred for g/w macs
-		 */
-		if (remote_ri && !CHECK_FLAG(flags, ZEBRA_MACIP_TYPE_GW)) {
-			uint32_t cur_seqnum;
-
-			/* Add MM extended community to route. */
-			cur_seqnum = mac_mobility_seqnum(remote_ri->attr);
-			add_mac_mobility_to_attr(cur_seqnum + 1, attr);
-		}
+	/* For non-GW MACs, update MAC mobility seq number, if needed. */
+	if (seq && !CHECK_FLAG(flags, ZEBRA_MACIP_TYPE_GW))
+		add_mac_mobility_to_attr(seq, attr);
```

Comment:
Does seq number needs to be incremented here ?

Reply:
The MAC mobility computation has moved to zebra because that is where all the relevant information is for inheritance between MAC and IP. BGP/EVPN code is only filling in the MAC Mobility extended community at the right time.